File: services/ordering/src/main/java/io/wkrzywiec/fooddelivery/ordering/domain/outgoing/OrderCreated.java
package io.wkrzywiec.fooddelivery.ordering.domain.outgoing;

import io.wkrzywiec.fooddelivery.commons.event.IntegrationMessageBody;
import io.wkrzywiec.fooddelivery.commons.model.Item;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

public record OrderCreated(UUID orderId, int version, String customerId, String farmId, String address, List<Item> items, BigDecimal deliveryCharge, BigDecimal total) implements IntegrationMessageBody {
}

File: services/ordering/src/main/kotlin/io/wkrzywiec/fooddelivery/ordering/domain/outgoing/OrderCreated.kt
package io.wkrzywiec.fooddelivery.ordering.domain.outgoing

import io.wkrzywiec.fooddelivery.commons.event.IntegrationMessageBody
import io.wkrzywiec.fooddelivery.commons.model.Item
import java.math.BigDecimal
import java.util.*

data class OrderCreated(
    val id: UUID,
    val aggregateVersion: Int,
    val customerId: String,
    val farmId: String,
    val address: String,
    val items: List<Item>,
    val deliveryCharge: BigDecimal,
    val total: BigDecimal
) : IntegrationMessageBody {
    override fun orderId(): UUID = id
    override fun version(): Int = aggregateVersion
}
